### PR TITLE
feat: added support for icomoon icons

### PR DESCRIPTION
**Pull request by ayadav-eightfold**
Added support for icomoon icons

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -125,6 +125,7 @@
         "html-replace-webpack-plugin": "2.5.6",
         "html-webpack-plugin": "5.5.0",
         "husky": "7.0.4",
+        "icomoon-react": "^3.0.0",
```
Comment (ayadav-eightfold) [inline package.json:128]: Remove "^"

File: src/components/ConfigProvider/ConfigProvider.types.ts [hunk 2]
```diff
@@ -18,4 +19,9 @@ export interface ConfigProviderProps {
      * @default { focusVisible: true, focusVisibleElement: document.documentElement }
      */
     focusVisibleOptions?: FocusVisibleOptions;
+    /**
+     * selection.json as generated by icomoon for icons
+     * @default {}
+     */
+    icomoonIconSet?: object | null;
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.types.ts:26]:
```suggestion
    icomoonIconSet?: Object;
```

File: src/components/Icon/selection.json [hunk 3]
```diff
@@ -0,0 +1,183 @@
+{
+    "IcoMoonType": "selection",
```
Comment (ychhabra-eightfold) [inline src/components/Icon/selection.json:2]: I hope this is just a sample selection.json.

File: src/components/ConfigProvider/ConfigProvider.types.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ export interface IConfigContext {
     themeOptions: ThemeOptions;
     setThemeOptions: (themeOptions: ThemeOptions) => void;
     registeredTheme?: IRegisterTheme;
+    icomoonIconSet?: object | null;
```
Comment (ychhabra-eightfold) [inline src/components/ConfigProvider/ConfigProvider.types.ts:8]:
```suggestion
    icomoonIconSet?: Object;
```

File: src/components/Icon/Icon.types.ts [hunk 5]
```diff
@@ -9,11 +9,11 @@ export enum IconSize {
     XSmall = '14px',
 }
 
-export interface IconProps extends OcBaseProps<HTMLSpanElement>, MdiIconProps {
+export interface IconProps extends OcBaseProps<HTMLSpanElement> {
```
Comment (ychhabra-eightfold) [inline src/components/Icon/Icon.types.ts:12]: Can we take two different props approach here.
https://stackoverflow.com/questions/58630750/react-typescript-component-with-two-different-prop-interfaces

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Icon/selection.json (hunk 3 above)
Comment (ayadav-eightfold) [reply src/components/Icon/selection.json:2]: Yes it is, the actual icon set needs to be passed through config provider

File: src/components/Icon/Icon.tsx [hunk 6]
```diff
@@ -20,12 +23,37 @@ export const Icon: FC<IconProps> = ({
...
+    icomoonIconName,
 }) => {
+    const { icomoonIconSet } = useConfig();
+
     const iconClassNames: string = mergeClasses([
         classNames,
         styles.iconWrapper,
     ]);
 
+    const IconComponent = icomoonIconName ? (
```
Comment (ychhabra-eightfold) [inline src/components/Icon/Icon.tsx:35]:
```suggestion
    const iconComponent = icomoonIconName ? (
```
nit

**Review by ychhabra-eightfold — APPROVED** [review]

**Pull request merged**